sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2185,8 +2185,9 @@ cl_int enqueueImpKernel(
     // If we have got non-success error code, let's analyze it to emit nice
     // exception explaining what was wrong
     const device_impl &DeviceImpl = *(Queue->getDeviceImplPtr());
-    return detail::enqueue_kernel_launch::handleError(Error, DeviceImpl, Kernel,
+    detail::enqueue_kernel_launch::handleErrorOrWarning(Error, DeviceImpl, Kernel,
                                                       NDRDesc);
+    return Error;
```

Comment:
Should we consider the error as handled here? That is, since `handleErrorOrWarning` should have performed the appropriate actions for warning, which is the only case that reaches this, can we consider it a success from here?

Reply:
I suppose we can think of the Error or warning as being handled, so we can delete this return statement